XenAdmin/Wizards/NewCPUGroupWizard/NewCPUGroupCPUCoresPage.Designer.cs

```diff
@@ -0,0 +1,175 @@
+            this.tableLayoutPanel1.SetColumnSpan(this.dataGridView1, 3);
+            this.dataGridView1.EditMode = System.Windows.Forms.DataGridViewEditMode.EditProgrammatically;
+            this.dataGridView1.GridColor = System.Drawing.SystemColors.Window;
+            this.dataGridView1.MultiSelect = false;
+            this.dataGridView1.Name = "dataGridView1";
+            this.dataGridView1.ReadOnly = true;
+            this.dataGridView1.RowHeadersVisible = false;
+            this.dataGridView1.SelectionMode = System.Windows.Forms.DataGridViewSelectionMode.FullRowSelect;
+            this.dataGridView1.ShowCellErrors = false;
+            this.dataGridView1.ShowEditingIcon = false;
+            this.dataGridView1.ShowRowErrors = false;
+            this.dataGridView1.StandardTab = true;
+            // 
+            // CheckBoxColumn
+            // 
+            resources.ApplyResources(this.CheckBoxColumn, "CheckBoxColumn");
+            this.CheckBoxColumn.Name = "CheckBoxColumn";
+            this.CheckBoxColumn.ReadOnly = true;
+            // 
+            // NameColumn
+            // 
+            resources.ApplyResources(this.NameColumn, "NameColumn");
+            this.NameColumn.Name = "NameColumn";
+            this.NameColumn.ReadOnly = true;
+            // 
+            // DescriptionColumn
+            // 
+            resources.ApplyResources(this.DescriptionColumn, "DescriptionColumn");
+            this.DescriptionColumn.Name = "DescriptionColumn";
+            this.DescriptionColumn.ReadOnly = true;
+            // 
+            // CurrentCPUGroupColumn
+            // 
+            resources.ApplyResources(this.CurrentCPUGroupColumn, "CurrentCPUGroupColumn");
+            this.CurrentCPUGroupColumn.Name = "CurrentCPUGroupColumn";
+            this.CurrentCPUGroupColumn.ReadOnly = true;
+            // 
+            // NewCPUGroupCPUCoresPage
+            // 
+            resources.ApplyResources(this, "$this");
+            this.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Dpi;
+            this.Controls.Add(this.tableLayoutPanel1);
+            this.Name = "NewCPUGroupCPUCoresPage";
+            this.tableLayoutPanel1.ResumeLayout(false);
+            this.tableLayoutPanel1.PerformLayout();
+            ((System.ComponentModel.ISupportInitialize)(this.dataGridView1)).EndInit();
+            this.ResumeLayout(false);
+
+        }
+
+        #endregion
+
+        protected System.Windows.Forms.TableLayoutPanel tableLayoutPanel1;
+        private System.Windows.Forms.Label label2;
+        private Controls.SearchTextBox searchTextBox1;
+        private System.Windows.Forms.DataGridView dataGridView1;
+        private System.Windows.Forms.Label label1;
+        private System.Windows.Forms.Label labelCounterVMs;
+        private System.Windows.Forms.Button buttonSelectAll;
+        private System.Windows.Forms.Button buttonClearAll;
+        private System.Windows.Forms.DataGridViewTextBoxColumn CheckBoxColumn;

```

Comment:
From what I see on the design, we also need a column for the expand/collapse icon. Now the design is not finalised yet: do we want the headers visible or hidden?

Reply:
According to other dialogs contain the expand/collapse icon(e.g. Attach Disk Dialog), they don't add  a column specially for the icon. So, I am wondering if it is necessary.

Reply:
The tree on the Attach disk dialog is a different control, it is a CustomTreeView <=FlickerFreeListBox <=ListBox. Here you have used a DataGridView, in which case you'll need a separate column. Have a look, for example, on the RollingPoolUpgrade wizard or the ManageUpdates Page to see how these columns work.

Reply:
Ok, I have changed it. I'm sorry to say that I'm both a fresh to XenCenter and C#.